src/public/modules/forms/helpers/CsvMergedHeadersGenerator.ts

```diff
@@ -1,21 +1,32 @@
-const moment = require('moment-timezone')
-const keyBy = require('lodash/keyBy')
-const { CsvGenerator } = require('./CsvGenerator')
-const { getResponseInstance } = require('./response-factory')
-
-/**
- * @typedef {{
- *    _id: string,
- *    question: string,
- *    answer?: string,
- *    answerArray?: string[],
- *    fieldType: string,
- *    isHeader?: boolean,
- * }} DisplayedResponse
- */
-
-class CsvMergedHeadersGenerator extends CsvGenerator {
-  constructor(expectedNumberOfRecords, numOfMetaDataRows) {
+import { keyBy } from 'lodash'
+import moment from 'moment-timezone'
+import { Dictionary } from 'ts-essentials/dist/types'
+
+import { DisplayedResponse } from 'src/types/response'
+
+import { Response } from './csv-response-classes'
+import { CsvGenerator } from './CsvGenerator'
+import { getResponseInstance } from './response-factory'
+
+type UnprocessedRecord = {
+  created: string
+  submissionId: string
+  record: ReturnType<typeof keyBy>
```

Comment:
haha we're going to have to be more specific than this - have you checked out the documentation for `keyBy`? if you actually hover your mouse over `ReturnType<typeof keyBy>`, you'll notice that it's `any` :P